engine/src/flutter/shell/platform/windows/host_window.h

```diff
@@ -60,17 +84,23 @@ class HostWindow {
   // Set the fullscreen state. |display_id| indicates the display where
   // the window should be shown fullscreen; std::nullopt indicates
   // that no display was specified, so the current display may be used.
-  void SetFullscreen(bool fullscreen,
-                     std::optional<FlutterEngineDisplayId> display_id);
+  virtual void SetFullscreen(bool fullscreen,
+                             std::optional<FlutterEngineDisplayId> display_id);
 
   // Returns |true| if this window is fullscreen, otherwise |false|.
-  bool GetFullscreen() const;
+  virtual bool GetFullscreen() const;
 
   // Given a window identifier, returns the window content size of the
   // window.
   static ActualWindowSize GetWindowContentSize(HWND hwnd);
 
- private:
+  // Returns the owner window, or nullptr if none.
+  HostWindow* GetOwnerWindow() const;
+
+  // Processes modal state update for single layer of window hierarchy.
```

Comment:
Could you expand what this method does?

From my understanding, this is called recursively when a child dialog window is created or destroyed. This starts from the root and works its way down to the child dialog window (or its parent) to ensure the windows on this path have the correct enabled/disabled state. Is that correct?

If so, could you add more docs explaining this?